lib/src/components/NxTable/__tests__/NxTable.test.tsx

```diff
@@ -5,162 +5,448 @@
+                <NxTable.Row clickAccessibleLabel="dolphin">
+                  <NxTable.Cell chevron/>
+                </NxTable.Row>
+              </NxTable.Body>
+            </NxTable>
+          );
 
-  it('correctly updates the colspan if the number of columns changes', async function() {
-    const component = mount(
-      <NxTable>
-        <NxTableHead>
-          <NxTableRow>
-            <NxTableCell>Foo</NxTableCell>
-            <NxTableCell>Bar</NxTableCell>
-          </NxTableRow>
-        </NxTableHead>
-        <NxTableBody error="Error!"/>
-      </NxTable>,
-      { attachTo: mountContainer }
-    );
+          expect(screen.getByRole('button')).toHaveAccessibleName('dolphin');
+        });
+      });
+
+      describe('when isHeader', function() {
+        it('renders a header cell with the text "Select Row"', function() {
+          const component = renderColumnHeader({
+            chevron: true,
+            sortDir: 'asc',
+            children: <span>foo</span>
+          })!;
+
+          expect(component).toHaveTextContent('Select Row');
+        });
+      });
+    });
+
+    describe('when clickable and the rowBtnIcon is set', function() {
+      const renderTableCell = (cellProps?: NxTableCellProps, rowProps?: NxTableRowProps, isHeader?: boolean) => {
+        const children = (
+          <NxTable.Row isClickable={true} { ...rowProps }>
+            <NxTable.Cell>Foo</NxTable.Cell>
+            <NxTable.Cell>Bar</NxTable.Cell>
+            <NxTable.Cell { ...cellProps }>Boo</NxTable.Cell>
+          </NxTable.Row>
+        );
+
+        return render(
+          <NxTable>
+            { isHeader ? <NxTable.Head>{children}</NxTable.Head>
+            : <NxTable.Body>{children}</NxTable.Body>
+            }
+          </NxTable>
+        ).getAllByRole(isHeader ? 'columnheader' : 'cell')[2] as HTMLElement;
+      };
+
+      describe('when isHeader', function() {
+        it('renders a header cell with the text "Select Row"', function() {
+          const component = renderTableCell({
+            rowBtnIcon: faEdit,
+            sortDir: 'asc'
+          }, undefined, true)!;
+
+          expect(component).toHaveTextContent('Select Row');
+        });
+      });
+
+      describe('when not isHeader', function() {
+        it('does not show button when rowBtnIcon is undefined', function() {
+          expect(within(renderTableCell({ rowBtnIcon: undefined })).queryByRole('button')).not.toBeInTheDocument();
+        });
+
+        it('does not show button when rowBtnIcon is null', function() {
+          expect(within(renderTableCell({ rowBtnIcon: null })).queryByRole('button')).not.toBeInTheDocument();
+        });
+
+        it('show button when rowBtnIcon faEdit is set', function() {
+          expect(within(renderTableCell({ rowBtnIcon: faEdit })).getByRole('button')).toBeInTheDocument();
+        });
+
+        it('ignores the JSX children and sort settings and renders a button instead',
```

Comment:
Considering that sort headers _also_ render buttons, is this test adequate? Maybe we should assert that the accessible name of the button doesn't include "Boo" and also doesn't include "ascending"

Reply:
> Considering that sort headers _also_ render buttons, is this test adequate? Maybe we should assert that the accessible name of the button doesn't include "Boo" and also doesn't include "ascending"

Thanks, let me know if this works: 608d1323ce1366993ed3dae2d77c45402510ba4f